### ISSUE TITLE
Console warnings are created when plotting charts

### ISSUE THREAD

**Issue by ThomUK** [nhs-r-community/NHSRplotthedots#17]
When plotting using spc(), some ggplot console warnings are issued, which are potentially unsettling to users and probably easy to fix.  

To reproduce:
```
values <- 1:12
dates <- seq.Date(from = as.Date("2020-01-01"), by = "month", length.out = 12)
dtf <- tibble(values, dates)

NHSRplotthedots::spc(dtf, "values", "dates")
```

This results in the console warnings:
```
Warning messages:
1: Removed 12 row(s) containing missing values (geom_path). 
2: Removed 12 row(s) containing missing values (geom_path). 
3: Removed 6 rows containing missing values (geom_point). 
4: Removed 6 rows containing missing values (geom_point). 
```

The geom_path warnings are the result of the target line and trajectory line being unused in this plot (ie. NAs in the dataframe columns).
The geom_point warnings are the result of the unplotted orange and blue special cause dots (again, NAs in those dataframe columns).

**Comment by tomjemmett** [comment]
this (usually) can be solved with adding the na.rm = TRUE argument to geom*, though my guess is most of these are coming from the calculate point highlighting function - because we are dealing with lag/lead, we introduce NA values there... as part of #48 I'm looking at splitting this function into lots of subfunctions that will be easier to test and validate which may close this issue

**Comment by tomjemmett** [comment]
I have resolved this in #48 by adding in `na.rm = TRUE` to certain `geom_` calls, namely the optional fields like target, trajectory, and the points that are highlighted

**Comment by ThomUK** [comment]
That's good to know thanks, I'll assign this to me and make a quick PR for it later tonight.
